metals/src/main/scala/scala/meta/internal/metals/BuildServerConnection.scala

```diff
@@ -47,6 +57,12 @@ class BuildServerConnection private (
   // the name is set before when establishing conenction
   def name: String = initialConnection.socketConnection.serverName
 
+  def onReconnection(
+      index: BuildServerConnection => Future[Unit]
+  ): Unit = {
+    onReconnection.set(index)
```

Comment:
We need to reindex the workspace and this is to make sure that we will only try to reindex once we finished the indexing at least once.